pkg/es/config/config.go

```diff
@@ -154,7 +155,19 @@ func (c *Configuration) NewClient(logger *zap.Logger, metricsFactory metrics.Fac
 	if err != nil {
 		return nil, err
 	}
-	return eswrapper.WrapESClient(rawClient, service), nil
+
+	// Determine ElasticSearch Version
+	pingResult, _, err := rawClient.Ping(c.Servers[0]).Do(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	esVersion, err := strconv.Atoi(string(pingResult.Version.Number[0]))
+	if err != nil {
+		return nil, err
+	}
+	logger.Sugar().Infof("Detected elasticsearch %d", esVersion)
```

Comment:
Change this to structured logging: `logger.Info("Elasticsearch detected", zap.Int("version", esVersion))`